airflow/providers/amazon/aws/hooks/redshift_sql.py

```diff
@@ -104,7 +104,11 @@ def get_iam_token(self, conn: Connection) -> tuple[str, str, int]:
         port = conn.port or 5439
         # Pull the custer-identifier from the beginning of the Redshift URL
         # ex. my-cluster.ccdre4hpd39h.us-east-1.redshift.amazonaws.com returns my-cluster
-        cluster_identifier = conn.extra_dejson.get("cluster_identifier", conn.host.split(".")[0])
+        cluster_identifier = conn.extra_dejson.get("cluster_identifier")
+        if not cluster_identifier and conn.host:
+            cluster_identifier = conn.host.split(".", 1)[0]
+        if not cluster_identifier:
+            raise Exception("Please set cluster_identifier or host in redshift connection.")
```

Comment:
Nit
```suggestion
        if not cluster_identifier:
            if conn.host:
                cluster_identifier = conn.host.split(".", 1)[0]
            else:
                raise Exception("Please set cluster_identifier or host in redshift connection.")
```

Reply:
Changed it [here](https://github.com/apache/airflow/pull/31567/commits/0b106d459f353195f8dd0df6ed5954e54adeea48)